securedrop_client/gui/widgets.py

```diff
@@ -1807,6 +1803,208 @@ def _on_left_click(self):
+        self.usb_error_message.setWordWrap(True)
+        usb_instructions = SecureQLabel(_(
+            'Please insert your encrypted drive into one of the USB ports marked EXTERNAL.'))
+        usb_instructions.setWordWrap(True)
+        buttons = QWidget()
+        buttons_layout = QHBoxLayout()
+        buttons.setLayout(buttons_layout)
+        usb_cancel_button = QPushButton(_('CANCEL'))
+        retry_export_button = QPushButton(_('CONTINUE'))
+        buttons_layout.addWidget(usb_cancel_button)
+        buttons_layout.addWidget(retry_export_button)
+        usb_form_layout.addWidget(self.usb_error_message)
+        usb_form_layout.addWidget(usb_instructions)
+        usb_form_layout.addWidget(buttons, alignment=Qt.AlignRight)
+
+        # Passphrase Form
+        self.passphrase_form = QWidget()
+        self.passphrase_form.setObjectName('passphrase_form')
+        passphrase_form_layout = QVBoxLayout()
+        self.passphrase_form.setLayout(passphrase_form_layout)
+        self.passphrase_error_message = SecureQLabel(_(
+            'The passphrase provided did not work. Please try again.'))
+        self.passphrase_error_message.setWordWrap(True)
+        passphrase_instructions = SecureQLabel(_('Enter password for safe USB drive.'))
+        passphrase_instructions.setWordWrap(True)
+        passphrase_label = SecureQLabel(_('Passphrase'))
+        passphrase_label.setObjectName('passphrase_label')
+        self.passphrase_field = QLineEdit()
+        self.passphrase_field.setEchoMode(QLineEdit.Password)
+        buttons = QWidget()
+        buttons_layout = QHBoxLayout()
+        buttons.setLayout(buttons_layout)
+        passphrase_cancel_button = QPushButton(_('CANCEL'))
+        unlock_disk_button = QPushButton(_('SUBMIT'))
+        buttons_layout.addWidget(passphrase_cancel_button)
+        buttons_layout.addWidget(unlock_disk_button)
+        passphrase_form_layout.addWidget(self.passphrase_error_message)
+        passphrase_form_layout.addWidget(passphrase_instructions)
+        passphrase_form_layout.addWidget(passphrase_label)
+        passphrase_form_layout.addWidget(self.passphrase_field)
+        passphrase_form_layout.addWidget(buttons, alignment=Qt.AlignRight)
+        self.passphrase_error_message.hide()
+
+        layout.addWidget(self.starting_export_message)
+        layout.addWidget(self.generic_error)
+        layout.addWidget(self.insert_usb_form)
+        layout.addWidget(self.passphrase_form)
+
+        self.generic_error.hide()
+        self.insert_usb_form.hide()
+        self.passphrase_form.hide()
+
+        usb_cancel_button.clicked.connect(self.close)
+        passphrase_cancel_button.clicked.connect(self.close)
+        retry_export_button.clicked.connect(self._on_retry_export_button_clicked)
+        unlock_disk_button.clicked.connect(self._on_unlock_disk_clicked)
+
+        self._export()
+
+    def _export(self):
+        try:
+            self.controller.run_export_preflight_checks()
```

Comment:
@creviera detected an issue when testing this PR with the dialog showing "preparing export" not immediately popping up in some scenarios. this can happen due to this `run_export_preflight_checks()` call and the other export-related call `export_file_to_usb_drive()` executing in the main (GUI) thread. Since they are blocking calls and may potentially be long running, they can prevent the processing of events in the main GUI thread, meaning that the dialog might not pop up until after the blocking call returns. if we call `QApplication.processEvents()` after the show action to make sure that the action to ensure the GUI events we want to happen are processed prior to the beginning of the subsequent blocking call, then this issue with the dialog popping up doesn't happen. i am cool with resolving in a followup in this sprint (because adding `processEvents` like [this](https://github.com/freedomofpress/securedrop-client/commit/1a4b68efc7e46e2666bc30f64f6b52e3775af08d) should really only be a temporary fix) or as a commit in this PR.